src/GeekLearning.Storage.Azure/AzureStore.cs

```diff
@@ -113,7 +78,7 @@ public AzureStore(string storeName, string connectionString, string containerNam
             blockBlob.Properties.ContentType = contentType;
             blockBlob.Properties.CacheControl = "max-age=300, must-revalidate";
             await blockBlob.SetPropertiesAsync();
-            return new Internal.AzureFileReference(blockBlob);
+            return new Internal.AzureFileReference(blockBlob, withMetadata: true);
```

Comment:
Oh, you're right, didn't think of using this methods with existing Blobs...
There is already an issue in the current version with the properties being overwritten so.

Maybe we should check it the blob exists, and auto fetch properties in this case?